python/cm/migrations/0092_auto_20220906_0957.py

```diff
@@ -0,0 +1,183 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generated by Django 3.2.13 on 2022-09-06 09:57
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import migrations, models
+
+MODEL_TYPE = {
+    "adcm": "adcm",
+    "cluster": "cluster",
+    "clusterobject": "service",
+    "servicecomponent": "component",
+    "hostprovider": "provider",
+    "host": "host",
+}
+
+
+def _get_host_action_context(apps, obj, task):
+    Prototype = apps.get_model("cm", "Prototype")
+    ClusterObject = apps.get_model("cm", "ClusterObject")
+    ServiceComponent = apps.get_model("cm", "ServiceComponent")
+    cluster = obj.cluster
+    context = {"cluster": {"id": cluster.pk, "name": cluster.name}}
+    try:
+        if task.action.prototype.type == "service":
+            service_prototype = Prototype.objects.get(
+                type=task.action.prototype.type,
+                name=task.action.prototype.name,
+                bundle=cluster.prototype.bundle,
+                version=cluster.prototype.version,
+                parent=None,
+            )
+            service = ClusterObject.objects.get(prototype=service_prototype, cluster=cluster)
+            context["service"] = {"id": service.pk, "name": service.prototype.display_name}
+        elif task.action.prototype.type == "component":
+            service_prototype = Prototype.objects.get(
+                type=task.action.prototype.parent.type,
+                name=task.action.prototype.parent.name,
+                bundle=cluster.prototype.bundle,
+                version=cluster.prototype.version,
+                parent=None,
+            )
+            service = ClusterObject.objects.get(prototype=service_prototype, cluster=cluster)
+            component_prototype = Prototype.objects.get(
+                type=task.action.prototype.type,
+                name=task.action.prototype.name,
+                bundle=cluster.prototype.bundle,
+                version=cluster.prototype.version,
+                parent=service_prototype,
+            )
+            component = ServiceComponent.objects.get(
+                cluster=cluster,
+                service=service,
+                prototype=component_prototype,
+            )
+            context["service"] = {"id": service.pk, "name": service.prototype.display_name}
+            context["component"] = {
+                "id": component.pk,
+                "name": component.prototype.display_name,
+            }
+    except ObjectDoesNotExist:
```

Comment:
`try` block should be as short as possible, it this case we can divide it to some `try` blocks

Reply:
Fixed